### PR TITLE
extension mode configurable

### PR DESCRIPTION
**Pull request by qiaozha**
This is to solve 
1. issue https://github.com/Azure/autorest.az/issues/351 configurable extension
2. issue https://github.com/Azure/autorest.az/issues/361 wait command should exist when no show command in the group 
3. issue https://github.com/Azure/autorest.az/issues/358 wait command custom function should have sdk_no_wait
4. part of issue https://github.com/Azure/autorest.az/issues/364 
4.1 if the operation already has parameter 'name', main resource option shouldn't be applied.
4.2 change wait_comamnd to the custom_wait_command since python sdk get operation is not always consistent with the show command parameter.
4.3 add parameter description in param.py for wait command.
4.4 allow change subgroup in the command.

### CODE REVIEW

File: src/test/scenarios/managed-network/output/src/managed-network/azext_managed_network/generated/custom.py [hunk 1]
```diff
@@ -267,7 +284,9 @@ def managed_network_managed_network_peering_policy_mesh_topology_update(instance
...
                                                           resource_group_name,
                                                           managed_network_name,
-                                                          policy_name):
-    return client.begin_delete(resource_group_name=resource_group_name,
-                               managed_network_name=managed_network_name,
-                               managed_network_peering_policy_name=policy_name)
+                                                          policy_name,
+                                                          no_wait=False):
+    return sdk_no_wait(no_wait, client.begin_delete,
+                                                   resource_group_name=resource_group_name,
```
Comment (zikalino) [inline src/test/scenarios/managed-network/output/src/managed-network/azext_managed_network/generated/custom.py:290]: there may be some python formatting lint issue here?

Comment (qiaozha) [reply src/test/scenarios/managed-network/output/src/managed-network/azext_managed_network/generated/custom.py:286]: I am sure this one will not cause the lint error. but please don't merge this PR for now

File: src/plugins/azgenerator/TemplateAzureCliCommands.ts [hunk 2]
```diff
@@ -37,13 +38,13 @@ export function GenerateAzureCliCommands(model: CodeModelAz): string[] {
                 output.push("        client_factory=" + cf_name + ")");
                 let groupinfos = model.CommandGroup_Name.split(' ');
                 let extraInfo = "";
-                if(groupinfos.length == 2) {
+                if(groupinfos.length == 2 && model.Extension_Mode == 'experimental') {
```
Comment (fengzhou-msft) [inline src/plugins/azgenerator/TemplateAzureCliCommands.ts:41]: Add similar logic when Extension_Mode is preview.

Comment (qiaozha) [reply src/plugins/azgenerator/TemplateAzureCliCommands.ts:41]: I thought preview don't need to add this `is_preview=True` at the end of the command?

Comment (fengzhou-msft) [reply src/plugins/azgenerator/TemplateAzureCliCommands.ts:41]: If not added, it means GA.

File: src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/custom.py [hunk 3]
```diff
@@ -123,3 +125,291 @@ def datafactory_get_git_hub_access_token(cmd, client,
...
+                                                  factory_name=factory_name,
+                                                  integration_runtime_name=integration_runtime_name)
+
+
+def datafactory_integration_runtime_stop(cmd, client,
+                                         resource_group_name,
+                                         factory_name,
+                                         integration_runtime_name,
+                                         no_wait=False):
+    return sdk_no_wait(no_wait, client.begin_stop,
```
Comment (MyronFanQiu) [inline src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/custom.py:394]: is it possible to have better format? :)

Comment (qiaozha) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/custom.py:394]: what kind of format does CLI need?

Comment (qiaozha) [reply src/test/scenarios/datafactory/output/src/datafactory/azext_datafactory/generated/custom.py]: Changed, please review again.

**Pull request merged**